tools/setup_helpers/dist_check.py

```diff
@@ -58,10 +71,46 @@ def should_build_ib():
         #   mlx5_2
         #   mlx5_1
         #   mlx5_0
-        return True
+        ib_util_found = True
     except Exception:
         # We just take all the exceptions here without affecting the build
-        return False
+        ib_util_found = False
+
+    lib_paths = list(filter(bool, [
+        "/usr/lib/",
+        "/usr/lib/x86_64-linux-gnu/",
+        "/usr/lib/powerpc64le-linux-gnu/",
+        "/usr/lib/aarch64-linux-gnu/",
+    ] + gather_paths([
+        "LIBRARY_PATH",
+    ]) + gather_paths([
+        "LD_LIBRARY_PATH",
+    ])))
+
+    include_paths = [
+        "/usr/include/",
+    ]
+
+    if IS_CONDA:
+        lib_paths.append(os.path.join(CONDA_DIR, "lib"))
+        include_paths.append(os.path.join(CONDA_DIR, "include"))
+
+    for path in lib_paths:
+        if path is None or not os.path.exists(path):
+            continue
+        ib_libraries = sorted(glob.glob(os.path.join(path, "libibverbs*")))
```

Comment:
I don't think this works on Windows but I also don't think we care?

Reply:
yeah, we don't care, but maybe it's good not to run this code in windows . Let me change that